wake/demo.wake

```diff
@@ -105,6 +100,22 @@ publish vivadoVsrcHooks =
   makeBlackBoxHook "loopback"     "{blockPIOSifiveRoot}/rtl/loopback/loopback.sv".makeSource,
   Nil
 
+def pioDriver =
+  def sourceFiles =
+    (sources "{blockPIOSifiveRoot}/drivers/metal" `.*.c`)
+    ++ (sources "{blockPIOSifiveRoot}/drivers/metal" `.*.h`)
+  makeDriverImplementation
+  ("sifive,pio-0.1.0", Nil)
+  "sifive"
+  "pio"
+  sourceFiles
+  ("{blockPIOSifiveRoot}/drivers/metal" ,Nil)
+  sourceFiles
```

Comment:
I'm rewriting this code slightly so that I can use it as an example in the walkthrough README. I'm basically going to follow the pattern that we use in a few other places where all the positional arguments to `makeDriverImplementation` will be simple variables defined above the function call, which allows us to more easily tell what each argument actually is.